Services/DataX.Config/DataX.Config/ConfigGeneration/Processor/S450_PrepareTransformFile.cs

```diff
@@ -0,0 +1,76 @@
+﻿// *********************************************************************
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License
+// *********************************************************************
+using Newtonsoft.Json;
+using DataX.Config.Utility;
+using DataX.Contract;
+using DataX.Flow.CodegenRules;
+using System.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using DataX.Config.ConfigDataModel;
+
+namespace DataX.Config.ConfigGeneration.Processor
+{
+    /// <summary>
+    /// Produce the transform/query file section
+    /// </summary>
+    [Shared]
+    [Export(typeof(IFlowDeploymentProcessor))]
+    public class PrepareTransformFile : ProcessorBase
+    {
+        public const string TokenName_TransformFile = "processTransforms";
+        public const string AttachmentName_CodeGenObject = "rulesCode";
+
+        private ConfigGenConfiguration Configuration { get; }
+        private IKeyVaultClient KeyVaultClient { get; }
+        private IRuntimeConfigStorage RuntimeStorage { get; }
+
+        [ImportingConstructor]
+        public PrepareTransformFile(IKeyVaultClient keyvaultClient, IRuntimeConfigStorage runtimeStorage, ConfigGenConfiguration conf)
+        {
+            KeyVaultClient = keyvaultClient;
+            RuntimeStorage = runtimeStorage;
+            Configuration = conf;
+        }
+
+        public override int GetOrder()
+        {
+            return 450;
+        }
+
+
+        public override async Task<string> Process(FlowDeploymentSession flowToDeploy)
+        {
+            var config = flowToDeploy.Config;
+            var guiConfig = config?.GetGuiConfig();
+            if (guiConfig == null)
+            {
+                return "no gui input, skipped.";
+            }
+            string queries = string.Join("\n", guiConfig.Process?.Queries);
+
+            string ruleDefinitions = RuleDefinitionGenerator.GenerateRuleDefinitions(guiConfig.Rules, config.Name);
+            RulesCode rulesCode = CodeGen.GenerateCode(queries, ruleDefinitions, config.Name);
+
+            Ensure.NotNull(rulesCode, "rulesCode");
+
+            // Save the rulesCode object for downstream processing
+            flowToDeploy.SetAttachment(AttachmentName_CodeGenObject, rulesCode);
+
+            var runtimeKeyVaultName = flowToDeploy.GetTokenString(PortConfigurationSettings.TokenName_RuntimeKeyVaultName);
+            Ensure.NotNull(runtimeKeyVaultName, "runtimeKeyVaultName");
+
+            var secretName = $"{config.Name}-transform";
+            Configuration.TryGet(Constants.ConfigSettingName_SparkType, out string sparkType);
+            var uriPrefix = (sparkType != null && sparkType == Constants.SparkTypeDataBricks) ? Constants.PrefixSecretScope : Constants.PrefixKeyVault;
+
+            var transformFileSecret = $"{uriPrefix}://{runtimeKeyVaultName}/{secretName}";
+            flowToDeploy.SetStringToken(TokenName_TransformFile, transformFileSecret);
```

Comment:
Where is this secret saved to keyvault? In general it is not a good practice to have the secret token set without saving the secret. If the secret save fails, we will have a non-existent key set in the token.

Reply:
Saving this secret to KV does happen in S610_GenerateTransformFile.cs. I made it on purpose. since if we don't have any job to deploy, we shouldn't save any file and even any secret.

Reply:
And the reason why I have to compose this secret url here is that the url is needed to generate the job config which is s600.

Reply:
updated to use the secret url which was saved from the prepare steps.